maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/TestClassPath.java

```diff
@@ -69,9 +70,9 @@ void avoidArtifactDuplicates( Set<Artifact> providerArtifacts )
                         && ( classifier1 == null ? classifier2 == null : classifier1.equals( classifier2 ) ) )
                 {
                     it.remove();
-                    if ( logger.isDebugEnabled() )
+                    if ( mojo != null && mojo.getLog().isDebugEnabled() )
```

Comment:
I have only several problems with this class:

1. We have AbstractSurefireMojo here.

2. We know what provider is being executed. What about to add a boolean to this method `isPlatformPrvicer` or enum instead of using null-checks in `if ( junitJupiterApi != null && junitJupiterEngine == null )`.

3. Why we do not align the versions of `junit-platform-launcher` dependency of our provider in classpath according to user's dependency version as well?

Reply:
Ad 1 and 2: I'm trying to move the logic into the `ProviderInfo::getProviderClasspath`. The "null-checks" are needed though.

Ad 3: See the second method named `adoptArtifactVersions`. It will care about aligning versions.